pkg/clustermesh/clustermesh.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clustermesh
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/kvstore/store"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+// IdentityChangeFunc is the function called on identity changes in a remote
+// cluster
+type IdentityChangeFunc func(modType ipcache.CacheModification, oldIPIDPair *identity.IPIdentityPair, newIPIDPair identity.IPIdentityPair)
```

Comment:
It would be nice to use the same signature as the `OnIPIdentityCacheChange` method of the `ipcache.IPIdentityMappingListener` interface, or even to use that interface directly.

Reply:
The type is actually no longer needed given your recent refactoring

```
                                ipCacheWatcher := ipcache.NewIPIdentityWatcher(backend)
                                go ipCacheWatcher.Watch()
```

The above takes care of everything. I've removed the type.